docs/user_guide.md

```diff
@@ -893,6 +893,19 @@ This example uses the same project and repositories as example 1, but sets the r
 
 In this case, all of the images in repository A are retained because they have been pulled in the last 7 days. None of the images in repositories B to E are retained, because none of them has been pulled in the last week. In this example, 100 images are retained, as opposed to 34 images in example 1.
 
+#### Tag Retention Rules and Native Docker Tag Deletion
+
+**WARNING**: Due to native Docker tag deletion behavior, there is an issue with the current retention policy implementation. If you have multiple tags that refer to the same SHA digest, and if a subset of these tags are deleted by a configured retention policy, all of the remaining tags are also deleted. This violates the retention policy. 
+
+For example, you have following tags, listed according to their push time, and all of them refer to the same SHA digest:
+
+- `harbor-1.8`, pushed 8/14/2019 12:00am
+- `harbor-release`, pushed 8/14/2019 03:00am
+- `harbor-nightly`, pushed 8/14/2019 06:00am
+- `harbor-latest`, pushed 8/14/2019 09:00am
+
+You configure a retention policy to retain the two latest tags that match `harbor-*`, so that `harbor-rc` and `harbor-latest` are deleted. However, since all tags refer to the same SHA digest, this also policy deletes the tags `harbor-1.8` and `harbor-release`.
```

Comment:
Hi Stuart, right now actually none of them are deleted precisely because of this.  My ticket wasn't very clear on this.  So all of them are actually retained instead of deleted because delete can't be rolled back.  So we retain more than the policy should